### PR TITLE
[Improvement] Handle encryption at rest errors

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This PR contains API changes from Data Model and Request Strategy, where errors are thrown up code paths that result in database access. 

## Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/241
- [x] https://github.com/wireapp/wire-ios-data-model/pull/1020
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/1304
- [x] https://github.com/wireapp/wire-ios-share-engine/pull/177

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Location.swift [hunk 1]
```diff
@@ -44,8 +44,12 @@ extension ConversationInputBarViewController {
 extension ConversationInputBarViewController: LocationSelectionViewControllerDelegate {
     func locationSelectionViewController(_ viewController: LocationSelectionViewController, didSelectLocationWithData locationData: LocationData) {
         ZMUserSession.shared()?.enqueue {
-            self.conversation.append(location: locationData)
-            Analytics.shared().tagMediaActionCompleted(.location, inConversation: self.conversation)
+            do {
+                try self.conversation.appendLocation(with: locationData)
+                Analytics.shared().tagMediaActionCompleted(.location, inConversation: self.conversation)
+            } catch {
+                // No op
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Location.swift:51]: should we add log here?

Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Location.swift]: Done

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.swift [hunk 2]
```diff
@@ -524,12 +524,14 @@ final class ConversationInputBarViewController: UIViewController,
...
-
-            if self.conversation.appendKnock() != nil {
+            do {
+                try self.conversation.appendKnock()
                 Analytics.shared().tagMediaActionCompleted(.ping, inConversation: self.conversation)
 
                 AVSMediaManager.sharedInstance().playKnockSound()
                 self.notificationFeedbackGenerator.notificationOccurred(.success)
+            } catch {
+                // No op
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.swift:534]: Log here as well for completeness?

Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.swift]: Done

**Review by typfel — APPROVED** [review]

**Comment by johnxnguyen** [comment]
@typfel I also made the "Message Processing" tag on by default so we can check logs for the warnings.

**Pull request merged**